bootstrap-conda

```diff
@@ -17,13 +17,24 @@ OPTIONAL_PACKAGES=
 SAGELIB_PACKAGES=
 SAGELIB_OPTIONAL_PACKAGES=
 DEVELOP_PACKAGES=
+UNCONSTRAINED_PACKAGES=
+CONSTRAINED_PACKAGES=
 
 for PKG_BASE in $(sage-package list --has-file distros/conda.txt --exclude _sagemath); do
     PKG_SCRIPTS=build/pkgs/$PKG_BASE
     SYSTEM_PACKAGES_FILE=$PKG_SCRIPTS/distros/conda.txt
     PKG_TYPE=$(cat $PKG_SCRIPTS/type)
     PKG_SYSTEM_PACKAGES=$(echo $(${STRIP_COMMENTS} $SYSTEM_PACKAGES_FILE))
     if [ -n "$PKG_SYSTEM_PACKAGES" ]; then
+        case "$PKG_SYSTEM_PACKAGES" in
+            *([-A-Za-z0-9_ ]))  # just a package name
+                UNCONSTRAINED_PACKAGES+=" $PKG_BASE"
```

Comment:
I think it's a fragile design to use "unconstrained version" as an indicator for including or excluding a certain dependency in the environment. It would prevent us from adding a version constraint to non-direct dependencies (although I have to admit I don't even understand the purpose of having conda.txt files for such dependencies) and forces us to specify a version constraint for those packages that do should end up.
Going with Python's 'explicit is better than implicit' slogan, I think it would be better to have a certain indicator file that says 'I'm a direct dependency, so include me please'.

Reply:
> It would prevent us from adding a version constraint to non-direct dependencies

To the contrary. It does the right thing -- namely including the dependency with the version constraint in the environment file -- for the non-direct dependencies.

Reply:
> design to use "unconstrained version" as an indicator for including or excluding a certain dependency in the environment.

That's not what it does. Packages are excluded by the minimizer from the environment file when they are implied by dependence. But we never exclude packages that have their own version constraints, so that the version constraint is not lost.

Reply:
There is no need for any of the version constraints you added in this PR (intrinsic need, as in "if you install an older version that it's known to fail"). Thus you use version constraints as a a signal to include the given package in the environment file. I think this is not a good use of version constraints.